mdpow/workflows/base.py

```diff
@@ -0,0 +1,166 @@
+        }
+    )
+        if csv_save_dir is not None:
+            
+            directory_paths.to_csv(f'{csv_save_dir}/dir_paths.csv', index=False)
+            logger.info(f'dir_paths saved under {csv_save_dir}')
+
+        else:
+            current_directory = os.system('pwd')
+            directory_paths.to_csv('dir_paths.csv', index=False)
+            logger.info(f'dir_paths saved under {current_directory}')
+
+    elif csv is not None:
+        locations = pd.read_csv(csv)
+        directory_paths = locations.sort_values(by=['molecule', 'resname', 'path']).reset_index(drop=True)
+
+    return directory_paths
+
+def directory_iteration(directory_paths, ensemble_analysis, **kwargs):
+    """Takes a :class:`pandas.DataFrame` created by
+       :func:`~mdpow.workflows.base.directory_paths`
+       and iterates over the project paths to implement
+       :func:`~mdpow.workflows.dihedrals.automated_dihedral_analysis`
+       for each project directory.
+
+       Compatibility with more automated analyses in development.
+
+       :keywords:
+
+       *directory_paths*
+           :class:`pandas.DataFrame` that provides paths to MDPOW projects
+
+       *ensemble_analysis*
+           name of the :class:`~mdpow.analysis.ensemble.EnsembleAnalysis`
+           that corresponds to the desired automated workflow module
+
+       *kwargs*
+           keyword arguments from :func:`~mdpow.workflows.dihedrals.automated_dihedral_analysis`
+
+           .. autodata:: mdpow.workflows.dihedrals.automated_dihedral_analysis
+
+       .. rubric:: Example
+
+       Typical Workflow::
+
+           directory_paths = directory_paths(parent_directory='/foo/bar/MDPOW_projects')
+           directory_iteration(directory_paths, ensemble_analysis='DihedralAnalysis', **kwargs)
+
+    """
+
+    try:
+        for row in directory_paths.itertuples():
+            molname = row.molecule
+            resname = row.resname
+            dirname = row.path
+
+            logger.info(f'starting {molname}')
+
+            registry.registry[ensemble_analysis](dirname=dirname, resname=resname, molname=molname, **kwargs)
+
+            logger.info(f'{molname} completed')
+
+    except KeyError as err:
+        msg = (f"Invalid ensemble_analysis {err}. An EnsembleAnalysis type that corresponds "
+                "to an existing automated workflow module must be input as a kwarg. "
+                "ex: ensemble_analysis='DihedralAnalysis'")
+        logger.error(f'{err} is an invalid selection')
+
+        raise KeyError(msg)
+
+    except TypeError as err:
+        msg = (f"Invalid ensemble_analysis {ensemble_analysis}. An EnsembleAnalysis type that "
+                "corresponds to an existing automated workflow module must be input as a kwarg. "
+                "ex: ensemble_analysis='DihedralAnalysis'")
+        logger.error(f'workflow module for {ensemble_analysis} does not exist yet')
+
+        raise TypeError(msg)
+
+    return logger.info('all analyses completed')
```

Comment:
Returning logger does not make sense.

I think we should be returning a collection of the data that's produced — eg all filenames. How will you know after this has run, what you got? It should _not_ require knowing a file naming scheme.

Reply:
For right now, please change to 
```python
logger.info('all analyses completed')
return
```
(and elsewhere, too)